workspaces/marketplace/plugins/marketplace/src/components/InstalledPlugins/InstalledPluginsTable.tsx

```diff
@@ -0,0 +1,216 @@
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import Tooltip from '@mui/material/Tooltip';
+
+import { processPluginsForDisplay } from '../../utils/pluginProcessing';
+
+export const InstalledPluginsTable = () => {
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const [currentSearchTerm, setCurrentSearchTerm] = useState<string>('');
+  const { count } = useInstalledPluginsCount();
+  const dynamicPluginInfo = useApi(dynamicPluginsInfoApiRef);
+  let data: DynamicPluginInfo[] = [];
+  const columns: TableColumn<DynamicPluginInfo>[] = [
+    {
+      title: 'Name',
+      field: 'name',
+      align: 'left',
+      width: '30ch',
+      defaultSort: 'asc',
+      headerStyle: {
+        textAlign: 'left',
+      },
+      cellStyle: {
+        textAlign: 'left',
+      },
+    },
+    {
+      title: 'Version',
+      field: 'version',
+      width: '54ch',
+      align: 'left',
+      headerStyle: {
+        textAlign: 'left',
+      },
+      cellStyle: {
+        textAlign: 'left',
+      },
+    },
+    {
+      title: 'Actions',
+      align: 'right',
+      width: '78px',
+      headerStyle: {
+        textAlign: 'left',
+      },
+      cellStyle: {
+        textAlign: 'left',
+      },
+      render: (_rowData: DynamicPluginInfo) => {
+        return (
+          <Box display="flex" gap={1}>
+            <Tooltip title="Edit">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  // TODO: Implement edit functionality
+                }}
+              >
+                <EditIcon />
+              </IconButton>
+            </Tooltip>
+            <Tooltip title="Uninstall">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  // TODO: Implement delete functionality
+                }}
+              >
+                <DeleteIcon />
+              </IconButton>
+            </Tooltip>
+            <Tooltip title="Export configurations (YAML)">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  // TODO: Implement download functionality
+                }}
+              >
+                <FileDownloadOutlinedIcon />
+              </IconButton>
+            </Tooltip>
+          </Box>
+        );
+      },
+      sorting: false,
+    },
+  ];
+  const fetchData = async (
+    query: Query<DynamicPluginInfo>,
+  ): Promise<QueryResult<DynamicPluginInfo>> => {
+    const {
+      orderBy = { field: 'name' },
+      orderDirection = 'asc',
+      page = 0,
+      pageSize = 5,
+      search = '',
+    } = query || {};
+
+    // Track current search term for conditional empty message
+    setCurrentSearchTerm(search);
+
+    try {
+      // for now sorting/searching/pagination is handled client-side
+      const installedPlugins = await dynamicPluginInfo.listLoadedPlugins();
```

Comment:
I know you just moved the code. But actual this code makes an api call every time the user changes the filter. But we can address this in a follow up PR.

Can you create a bug for this?